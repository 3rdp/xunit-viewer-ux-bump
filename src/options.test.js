import React from 'react'
import Options from './options'
import renderer from 'react-test-renderer'

test('renders options', () => {
  const props = {
    key: 'suites',
    label: 'Suites',
    active: false,
    count: 100,
    total: 100,
    counts: [{
      key: 'passed',
      type: 'passed',
      total: 10
    }],
    toggles: [{
      key: 'all',
      label: 'all',
      visible: true,
      expanded: true,
      raw: true
    }]
  }

  const tree = renderer.create(<Options {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
