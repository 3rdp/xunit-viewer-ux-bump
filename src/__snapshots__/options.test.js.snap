// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders options 1`] = `
<div
  className="options card is-inactive"
>
  <header
    className="card-header"
  >
    <button
      className="card-header-icon is-hidden-tablet"
      onClick={[Function]}
    >
      <span
        className="icon"
      >
        <i
          aria-hidden="true"
          className="fas fa-angle-down"
        />
      </span>
    </button>
    <div
      className="options-inputs"
    >
      <div
        className="field options-search"
      >
        <div
          className="control"
        >
          <input
            className="input"
            placeholder="Suites"
            type="text"
          />
        </div>
      </div>
      <div
        className="options-total"
      >
        <b>
          100
        </b>
        <span>
          /
          100
        </span>
      </div>
      <div
        className="options-count"
      >
        <span
          className="icon"
        >
          <i
            aria-hidden="true"
            className="fas fa-check"
          />
        </span>
        <span>
          10
        </span>
      </div>
    </div>
    <button
      className="card-header-icon is-hidden-mobile"
      onClick={[Function]}
    >
      <span
        className="icon"
      >
        <i
          aria-hidden="true"
          className="fas fa-angle-down"
        />
      </span>
    </button>
  </header>
  <div
    className="card-content"
  >
    <div
      className="toggles-container"
    >
      <div
        className="toggles"
      >
        <h6
          className="title is-6"
        >
          <span>
            All
          </span>
        </h6>
        <button
          className="button toggle is-active"
          disabled={true}
        >
          <div
            className="toggle-rail"
          >
            <div
              className="toggle-handle"
            />
          </div>
          <span
            className="icon"
          >
            <i
              className="fas fa-eye"
            />
          </span>
          <span>
            Visible
          </span>
        </button>
        <button
          className="button toggle is-active"
          disabled={true}
        >
          <div
            className="toggle-rail"
          >
            <div
              className="toggle-handle"
            />
          </div>
          <span
            className="icon"
          >
            <i
              className="fas fa-chevron-down"
            />
          </span>
          <span>
            Expanded
          </span>
        </button>
        <button
          className="button toggle is-active"
          disabled={true}
        >
          <div
            className="toggle-rail"
          >
            <div
              className="toggle-handle"
            />
          </div>
          <span
            className="icon"
          >
            <i
              className="fas fa-code"
            />
          </span>
          <span>
            Raw
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
